src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -116,6 +113,17 @@ private PackageSpec GetPackageSpec()
             return null;
         }
 
+        internal override async ValueTask<PackageSpec> GetPackageSpecAsync(CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            return await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
This seems overly complicated. 

Can't you rewrite the whole method with; 

> return Task.FromResult(GetPackageSpec());

Reply:
Sure. Rewrited.